electron/src/lib/CoreProtocol.test.main.ts

```diff
@@ -0,0 +1,29 @@
+import * as assert from 'assert';
+import sinon from 'sinon';
+import {CustomProtocolHandler} from './CoreProtocol';
+
+let protocolHandler: CustomProtocolHandler = undefined;
+
+describe('dispatcher', () => {
+  beforeEach(() => {
+    protocolHandler = new CustomProtocolHandler();
+  });
+
+  it('forwards conversation deep links to the webapp', () => {
+    sinon.replace(protocolHandler, 'forwardHashLocation', sinon.fake());
+    protocolHandler.dispatcher('wire://conversation/8cdb44a0-418b-4188-9a53-7c477a7848dd');
+    assert.strictEqual((protocolHandler['forwardHashLocation'] as sinon.SinonSpy).calledOnce, true);
+  });
+
+  it('forwards user profile deep links to the webapp', () => {
+    sinon.replace(protocolHandler, 'forwardHashLocation', sinon.fake());
```

Comment:
Mocking private function means you are not testing most of the logic.
What you want to mock would rather be the `WindowManager.sendActionToPrimaryWindow` method and check that it's really called.

Reply:
Yepp. I can mock `sendActionToPrimaryWindow` and do an expectation on the parameters that have been used to call it. Good point. Will mark my PR as draft.